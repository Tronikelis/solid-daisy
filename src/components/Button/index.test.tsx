import { render } from "@solidjs/testing-library";
import { describe, expect, it } from "vitest";

import { Button } from ".";

describe("<Button />", () => {
    it("renders correctly", () => {
        const { container } = render(() => <Button>Hi</Button>);
        expect(container).toMatchSnapshot();
    });

    it("loading works", () => {
        const screen = render(() => <Button loading data-testid="button" />);
        expect(screen.getByTestId("button")).toBeDisabled();
    });

    it("icons show up", () => {
        const screen = render(() => (
            <Button leftIcon={<div>left</div>} rightIcon={<div>right</div>} />
        ));

        expect(screen.getByText("left")).toBeInTheDocument();
        expect(screen.getByText("right")).toBeInTheDocument();
    });
});
