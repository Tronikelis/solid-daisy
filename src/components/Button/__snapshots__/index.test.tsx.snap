// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > renders correctly 1`] = `
<div>
  <button
    class="btn"
    type="button"
  >
    
    
    Hi
    
  </button>
</div>
`;
